Examine Blocks

13spacemen opened the pull request:
Adds Examine Blocks as seen on Skyrat and other servers, but tweaked a bit to look better

LIGHT MODE:
![13pxlight](https://user-images.githubusercontent.com/46101244/175790441-0d540b6a-71c2-4bab-9b74-01ff03cc4f4a.png)


DARK MODE:
![13pxdark](https://user-images.githubusercontent.com/46101244/175790450-d2ab8d4f-97ec-4f3a-aa59-b82a9a920207.png)




- [x] Add examine blocks to more places
- [x] Remove initial and final `*---------*` since that was a separator but is no longer needed
- [x] Tweak the colors a bit
- [x] Fix any bugs

Fixes https://github.com/tgstation/tgstation/issues/67983

## Why?
Helps multi-text outputs stand out from the rest of the chat, makes it way more readable when you use the various analyzers or examine someone or something

:cl: 13spacemen, Gandalf2k15
add: Examining and other chat outputs now display in blocks to make them easier to see
fix: AI and borgs no longer have their name show twice upon being examined
/:cl:

13spacemen committed:
adds examine_block class and a define for it
made some outputs in chat use examine_block

san7890 commented:
Any screenshots on it in Dark Mode?

ArcaneMusic commented:
VERY clean

timothymtorres commented:
This is amazing.  +1

timothymtorres commented on code/game/objects/items/devices/scanners/slime_scanner.dm:
```diff
@@ -51,4 +51,4 @@
 		to_render += "\n[span_notice("Core mutation in progress: [T.effectmod]")]\
 					  \n[span_notice("Progress in core mutation: [T.applied] / [SLIME_EXTRACT_CROSSING_REQUIRED]")]"
-	to_chat(user, to_render + "\n========================")
+	to_chat(user, examine_block(to_render + "\n========================"))
```
I saw you remove the `======` from the other parts of the code but why not here as well?

13spacemen replied:
workin on it

13spacemen committed:
add examine block to tip of the round
clean up some ------ and ***** seperators
added <hr> tags to divide sections
cleaned up botany plant analyzer text outputs

Farquaar commented:
How did we play this game for so many years without this?

13spacemen committed:
bullet points for reagents

LemonInTheDark commented on code/modules/mob/mob.dm:
```diff
@@ -533,7 +533,12 @@
 		result = examinify.examine(src) // if a tree is examined but no client is there to see it, did the tree ever really exist?
 
-	to_chat(src, "<span class='infoplain'>[result.Join("\n")]</span>")
+	if(result.len)
+		for(var/i = 1, i <= result.len, i++)
+			if(!findtext(result[i], "<hr>"))
+				result[i] += "\n"
+
+	to_chat(src, "<div class='examine_block'><span class='infoplain'>[result.Join()]</span></div>")
```
```suggestion
	for(var/i in 1 to length(result))
		if(!findtext(result[i], "<hr>"))
			result[i] += "\n"
	to_chat(src, "<div class='examine_block'><span class='infoplain'>[result.Join()]</span></div>")
```
for(var/i in x to y) is faster, nicer, and functionally equivalent if the list's size doesn't change
to remove the trailing new line, do length(result) - 1

13spacemen committed:
removes <hr> from mobs examines
fixes AIs and borgs having a double "That's Default Cyborg!"
removed some \n newlines
minor code edits

13spacemen committed:
removes all <hr>
bold names in get_examine_name()
cleaned up plant analyzer output formatting
adjust colors and margins of examine_block class
remove \a from borg examine()

LemonInTheDark commented:
if this code was ported from skyrat, could you credit the authors whos work you're using?

13spacemen commented:
> if this code was ported from skyrat, could you credit the authors whos work you're using?

the PR is https://github.com/Skyrat-SS13/Skyrat-tg/pull/8153, i think the author is @Gandalf2k15 , added as a credit

13spacemen committed:
remove max-width from css

LemonInTheDark commented:
that seems off, hm

Wallemations commented:
Think we could get a screenshot of it with different text sizes?

stylemistake commented on tgui/packages/tgui-panel/styles/goon/chat-dark.scss:
```diff
@@ -900,6 +900,13 @@ em {
   margin-left: 3em;
 }
 
+.examine_block {
+  background: #1b1c1e;
+  border: 1px solid #a4bad6;
+  margin: 7px;
+  padding: 8px 12px;
+}
+
```
You should use `em` units to write these styles, one `em` is equal to whatever font size in px you have set in chat settings, but try to stick to "one unit of margin is 0.5em" convention that we have in tgui.

Suggestion:

```suggestion
.examine_block {
  background: #1b1c1e;
  border: 1px solid #a4bad6;
  margin: 0.5em;
  padding: 0.5em 0.75em;
}
```

stylemistake commented on tgui/packages/tgui-panel/styles/goon/chat-light.scss:
```diff
@@ -933,6 +933,13 @@ h1.alert, h2.alert {
   margin-left: 3em;
 }
 
+.examine_block {
+  background: #f2f7fa;
+  border: 1px solid #111a27;
+  margin: 7px;
+  padding: 8px 12px;
+}
+
```
```suggestion
.examine_block {
  background: #f2f7fa;
  border: 1px solid #111a27;
  margin: 0.5em;
  padding: 0.5em 0.75em;
}
```

TrashDoxx commented:
I love it ❤️

13spacemen committed:
changed margin and padding units from px to em

13spacemen commented:
> Think we could get a screenshot of it with different text sizes?
21px
![21pxbox](https://user-images.githubusercontent.com/46101244/175790474-34ab7a5f-b134-4fcb-953b-a72c96646e0b.png)
10px
![10pxbox](https://user-images.githubusercontent.com/46101244/175790476-4fa82d65-24c5-42a2-966d-2ade17620132.png)
13px
![13pxdark](https://user-images.githubusercontent.com/46101244/175790477-15388a70-bfe2-48e9-a973-c2f84680d1af.png)

margins and padding use em now so they scale well with text size

13spacemen committed:
minor edit
